### PR TITLE
Text input component

### PR DESCRIPTION
**Pull request by acchiang**
image

### CODE REVIEW

File: src/components/Common.js
Comment (ianmah) [inline src/components/Common.js:51]: Which parts are we hoping to keep from CardLike?

File: src/components/Common.js
Comment (ianmah) [inline src/components/Common.js:48]: Thoughts on making a new file for `Inputs`?

File: src/components/Common.js
Comment (acchiang) [reply src/components/Common.js:51]: not sure yet, i am going to figure out how to implement the error message/state for text inputs

**Comment by acchiang** [comment]
image

**Comment by acchiang** [comment]
WIP: missing hover behaviour for Invalid input

File: src/components/TextInput.js
Comment (acchiang) [inline src/components/TextInput.js:47]: idk why but i can't get `:invalid` to work, lmk if anyone has ideas, this is my hacky solution for now

File: src/components/TextInput.js
Comment (acchiang) [inline src/components/TextInput.js:70]: yes, these props are ugly af, anyone know how to pass props directly into TextInputBox (that's a styled.input)? am dum

File: src/components/TextInput.js
Comment (panjenny0) [reply src/components/TextInput.js]: I think if you change `${p => p.theme.colors.primary}` to `${p.theme.colors.primary}` it should work. 
It's complaining that `${p => p.theme.colors.primary}` is an invalid property value

before
Screen Shot 2020-11-09 at 7 56 06 PM

after
Screen Shot 2020-11-09 at 7 56 55 PM

Comment (acchiang) [reply src/components/TextInput.js]: omg @panjenny0 wrinkliest brain in the wild west... thank u

**Comment by acchiang** [comment]
the example text inputs live in /charcuterie

**Review by panjenny0 — DISMISSED** [review]

**Comment by ianmah** [comment]
Does this work for `textarea` components as well?

File: src/theme/ThemeProvider.js
Comment (ianmah) [inline src/theme/ThemeProvider.js:10]: I don't really think we need two warning colours @jvssicawu can we just use one?

Comment (jvssicawu) [reply src/theme/ThemeProvider.js:10]: yee one is fine for livesite, Im not sure if we use secondary warning for application form though

Comment (panjenny0) [reply src/theme/ThemeProvider.js:10]: Seems like the hacker app uses `#F18383` though - should we keep it, or change it to `#EF6C6C` to stay consistent? They are quite close.
Screen Shot 2020-11-15 at 12 04 56 AM

Comment (acchiang) [reply src/theme/ThemeProvider.js:10]: i did it because of the _slightly_ different shades here, so i thought we might want two warning colours in general in the future! i'm happy to make the change (to one warning colour), @jvssicawu can you please re-confirm if that's ok? ty!!! 
image

**Comment by acchiang** [comment]
> Does this work for `textarea` components as well?
soz for dumb q, what is a textarea (is it a styled component we already have in the codebase, eg. our search bar in FAQ)?

File: src/theme/ThemeProvider.js
Comment (ianmah) [reply src/theme/ThemeProvider.js:10]: They look identical to me heH

**Comment by ianmah** [comment]
A text area is a larger text box https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea

**Comment by acchiang** [comment]
> A textarea is a larger text box https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea
> Maybe I should've called it a textarea "element"

Ah gotcha. No there is no textarea implementation yet, I intend to work on it right after this :)  
image

It's not implemented yet because of additional features such as the scroll bar, max char limit. LMK if you have suggestions/preferences on how I should do it (new component? extension? TBD)

File: src/theme/ThemeProvider.js
Comment (acchiang) [reply src/theme/ThemeProvider.js:10]: HAHA same (that's why i added it so late, i realized like last min) 
They're very similar but i checked the figma css and they be different 😢  Will wait on design sign-off but ya i agree i think 1 warning colour is fine

Comment (jvssicawu) [reply src/theme/ThemeProvider.js:10]: I'm not sure why there's two, I can check with Tiff but since the difference is so minute i dont think it was intentional. you can change to use just one for now imo

Comment (jvssicawu) [reply src/theme/ThemeProvider.js:10]: oh actually LMAO sorry i took a second look because my brain kinda slow at 1am but seems like its specifically for the hover/focus effect so yea its probably intentional actually

**Comment by ianmah** [comment]
I quite like how the Cardlike reusable css works, maybe we could do something similar for the text input and textarea

**Review by ianmah — APPROVED** [review]

**Comment by acchiang** [comment]
oops had to merge to get rid of conflicts. pls approve again! thanks :D

**Review by panjenny0 — APPROVED** [review]
:shipit:

**Pull request merged**
